libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -656,6 +656,15 @@ void CompilerUtils::convertType(Type const& _typeOnStack, Type const& _targetTyp
 		solAssert(_typeOnStack == _targetType, "Invalid type conversion requested.");
 		break;
 	}
+
+	// Check the conversion result fits in a range.
+	if (targetTypeCategory == Type::Category::Enum)
```

Comment:
@chriseth I moved the overflow checking; but I'm not sure if this is what you want.  I did not find an enum cleaning routine.